docs/managing-vendor-dependencies.md

```diff
@@ -0,0 +1,176 @@
+# Managing vendor dependencies
+
+[`@theforeman/vendor`](https://github.com/theforeman/foreman-js/blob/master/packages/vendor) is a set of formally supported 3rd-party node_modules to use in foreman and foreman plugins.
+
+For more information, see the [readme.md](https://github.com/theforeman/foreman-js/blob/master/packages/vendor/readme.md)
+
+## Add a vendor dependency
+
+There are a few steps needed to add a vendor dependency.
+For this example, we will add the `@patternfly/react-table` dependency.
+
+### Step 1: Install the new dependency in vendor-core
+
+Run the `add-to-vendor` command to have your package installed as a dependency of the `vendor-core` project.
+
+```sh
+npm run add-to-vendor -- <dependency-to-add>
+```
+
+In our case:
+
+```sh
+npm run add-to-vendor -- @patternfly/react-table
+```
+
+### Step 2: Make the dependency accessible for consumers
+
+Typically, a dependency should come with javascript file/s which export method/s to use.
+
+To build the javascript file/s into the `vendor.js` dist file, we need to add all the javascript files to the [`vendor-core/lib/modules.js`](https://github.com/theforeman/foreman-js/blob/master/packages/vendor-core/lib/modules.js) file.
+
+In our case with `@patternfly/react-table`, there is only one javascript file so we should just add `@patternfly/react-table` to the exported array in the [`vendor-core/lib/modules.js`](https://github.com/theforeman/foreman-js/blob/master/packages/vendor-core/lib/modules.js) file:
+
+```diff
+'react',
+'react-dom',
+'@patternfly/react-core',
++'@patternfly/react-table',
+'redux',
+```
+
+### Step 3: Provides s/css with the package
+
+In some cases, a package also provides css or scss files.
+To have the css available to the vendor consumers (foreman and it's plugins), we should add them to the [`vendor-core/scss/vendor-core.scss`](https://github.com/theforeman/foreman-js/blob/master/packages/vendor-core/scss/vendor-core.scss) so they would get compiled into the `vendor.css` dist file.
+
+When a package provides variables and mixins, we should add them to [`vendor-core/scss/variables.scss`](https://github.com/theforeman/foreman-js/blob/master/packages/vendor-core/scss/variables.scss) and [`vendor-core/scss/mixins.scss`](https://github.com/theforeman/foreman-js/blob/master/packages/vendor-core/scss/mixins.scss) files.
+They would get compiled into `@theforeman/vendor/scss/variables.scss` and `@theforeman/vendor/scss/mixins.scss` files.
+The consumers can import those files and reuse the package variables or mixins files.
+
+**In our cases with the `@patternfly/react-table`, there are no s/css files so we can skip this step.**
+
+### Step 4: Build, and test with the foreman or with a foreman plugin
+
+Once we have the package installed, listed in the `modules.js` file and the `.scss` files, we should be able to build and use the package from foreman or a foreman plugin.
+
+Build:
+```sh
+npm run build
+```
```

Comment:
Isn't `npm install` a step here before building?

Reply:
Also could be nice to say where to run these commands, is it the top-level or in a packages sub-directory?

Reply:
@sharvit, what about these comments?

Reply:
@MariaAga me and also @adamruzicka seem to have missed it.
@sharvit could you open a follow-up to adress these?